From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Wed, 11 Dec 2019 23:56:50 +0100
Subject: [PATCH] {BUG in 5.5-rc1} drivers/base/core: use <0 to detect errors
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Detected by gcc-framac:

    drivers/base/core.c: In function
    ‘device_link_add_missing_supplier_links’:
    drivers/base/core.c:527:151: error: ?: using integer constants in
    boolean context [-Werror=int-in-bool-context]
      527 |   if (!fwnode_call_int_op(dev->fwnode, add_links, dev))
          |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
---
 drivers/base/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/base/core.c b/drivers/base/core.c
index 678b253b11d4..da5bb4f00131 100644
--- a/drivers/base/core.c
+++ b/drivers/base/core.c
@@ -524,7 +524,7 @@ static void device_link_add_missing_supplier_links(void)
 	mutex_lock(&wfs_lock);
 	list_for_each_entry_safe(dev, tmp, &wait_for_suppliers,
 				 links.needs_suppliers)
-		if (!fwnode_call_int_op(dev->fwnode, add_links, dev))
+		if (fwnode_call_int_op(dev->fwnode, add_links, dev) < 0)
 			list_del_init(&dev->links.needs_suppliers);
 	mutex_unlock(&wfs_lock);
 }
-- 
